### ISSUE TITLE
Preventing the VM being detected as a VM

### ISSUE THREAD

**Issue by oskardotglobal** [winapps-org/winapps#99]
I've started working on incorporating https://github.com/zhaodice/qemu-anti-detection into the windows docker image.
However, I can't quite seem to get QEMU to compile inside docker.
See https://github.com/oskardotglobal/docker-qemu-anti-detection and https://github.com/oskardotglobal/docker-qemu-anti-detection/actions/runs/9178044026/job/25237029138

**Comment by LDprg** [comment]
@oskardotglobal why did you write all build commands using only on run command? Spliting them into seperate commands is probably better.

**Comment by oskardotglobal** [comment]
It's better for image size and build time reasons

**Comment by oskardotglobal** [comment]
> @oskardotglobal I checked your repo, I have the suspicion that the qemu version of qemu-docker and the vm hiding software are different.

That doesn't really matter here, the built version will be preferred over the installed one because of path precedence.

**Comment by oskardotglobal** [comment]
If you check the log make fails with `ERROR: Unrecognized host OS (uname -s reports 'Linux')`

**Comment by oskardotglobal** [comment]
Which doesn't make any sense to me because the image actually runs Debian Trixie

**Comment by oskardotglobal** [comment]
So the OS should be recognized as Debian

**Comment by LDprg** [comment]
@oskardotglobal this makes sense. Uname is not working inside docker. I found quite some stuff:
- https://github.com/docker-32bit/ubuntu/issues/12
- https://github.com/dischoen/docker_uname
-

**Comment by oskardotglobal** [comment]
Thanks, this helps a lot

**Comment by oskardotglobal** [comment]
However, the uname name comes from the runner which is actions in this case

**Comment by LDprg** [comment]
> However, the uname name comes from the runner which is actions in this case

Yeah, but you could overwrite it in the docker anyways. And I am not sure if its always the host uname. People reported different stuff, maybe its depenedent on the base image thats being used.

**Comment by oskardotglobal** [comment]
It couldn't be, because docker reuses the host kernel
which is .. yk, the entire point of using docker

**Comment by oskardotglobal** [comment]
Wait, I just realized that error doesn't make any sense
`uname -s` isn't supposed to output distro name

**Comment by oskardotglobal** [comment]
... The error comes from not having installed gcc

**Comment by LDprg** [comment]
@oskardotglobal sorry for the comment deletion was a small accident on my mobile phone.

**Comment by oskardotglobal** [comment]
Oh by the way this is done already, I'd just have to add the new base image in the windows image

**Comment by oskardotglobal** [comment]
Then the only problem would be applying the QEMU XML config

**Comment by oskardotglobal** [comment]
Seems we can do that with https://libvirt.org/manpages/virsh.html#domxml-to-native
